src/core/server/core_usage_data/core_usage_stats_client.ts

```diff
@@ -88,67 +105,124 @@ export class CoreUsageStatsClient {
-    ];
-    await this.updateUsageStats(counterFieldNames, EXPORT_STATS_PREFIX);
+    const counterFieldNames = [`allTypesSelected.${isAllTypesSelected ? 'yes' : 'no'}`];
+    await this.updateUsageStats(counterFieldNames, EXPORT_STATS_PREFIX, options);
   }
 
-  private async updateUsageStats(counterFieldNames: string[], prefix: string) {
+  private async updateUsageStats(
+    counterFieldNames: string[],
+    prefix: string,
+    { request }: BaseIncrementOptions
+  ) {
     const options = { refresh: false };
     try {
       const repository = await this.repositoryPromise;
+      const fields = [...this.getCommonFieldsToIncrement(request), ...counterFieldNames];
       await repository.incrementCounter(
         CORE_USAGE_STATS_TYPE,
         CORE_USAGE_STATS_ID,
-        counterFieldNames.map((x) => `${prefix}.${x}`),
+        fields.map((x) => `${prefix}.${x}`),
         options
       );
     } catch (err) {
       // do nothing
     }
   }
+
+  private getIsDefaultNamespace(request: KibanaRequest) {
+    const requestBasePath = this.basePath.get(request); // obtain the original request basePath, as it may have been modified by a request interceptor
+    const pathToCheck = this.basePath.remove(requestBasePath); // remove the server basePath from the request basePath
+    const matchResult = pathToCheck.match(SPACE_CONTEXT_REGEX); // Look for `/s/space-url-context` in the base path
+
+    if (!matchResult || matchResult.length === 0) {
+      return true;
+    }
+
+    // Ignoring first result, we only want the capture group result at index 1
+    const [, spaceId] = matchResult;
+
+    return spaceId === DEFAULT_NAMESPACE_STRING;
+  }
+
+  private getCommonFieldsToIncrement(request: KibanaRequest) {
+    const isKibanaRequest = getIsKibanaRequest(request);
+    const isDefaultNamespace = this.getIsDefaultNamespace(request);
+    const namespaceField = isDefaultNamespace ? 'default' : 'custom';
+    const counterFieldNames = [
+      'total',
+      `namespace.${namespaceField}.total`,
+      `namespace.${namespaceField}.kibanaRequest.${isKibanaRequest ? 'yes' : 'no'}`,
+    ];
+    return counterFieldNames;
+  }
+}
+
+function getAllCommonFields(prefix: string) {
+  return [
+    'total',
+    'namespace.default.total',
+    'namespace.default.kibanaRequest.yes',
+    'namespace.default.kibanaRequest.no',
+    'namespace.custom.total',
+    'namespace.custom.kibanaRequest.yes',
+    'namespace.custom.kibanaRequest.no',
+  ].map((x) => `${prefix}.${x}`);
 }
 
-function getIsKibanaRequest(headers?: Headers) {
+function getIsKibanaRequest({ headers }: KibanaRequest) {
   // The presence of these three request headers gives us a good indication that this is a first-party request from the Kibana client.
   // We can't be 100% certain, but this is a reasonable attempt.
-  return headers && headers['kbn-version'] && headers.origin && headers.referer;
+  return headers && headers['kbn-version'] && headers['user-agent'] && headers.referer;
 }
```

Comment:
I changed this method to check the `user-agent` header instead of the `origin` header. See the description for 4d9d7f8:

> First implementation made a best-effort attempt to determine
> whether or not a request was "first-party" (from the Kibana client)
> by checking "kbn-version", "origin", and "referer". If these three
> headers are all present, we thought it was very likely that the
> request was first-party. However, further testing after adding
> usage stats collection for additional SO APIs determined this to be
> inaccurate; some actual first-party requests do not include the
> "origin" header. I am not sure exactly why this is the case, but I
> have changed the check to use "user-agent" instead. This is another
> indicator that a request likely did not come from a programmatic
> consumer.

Reply:
Can you elaborate on why the presence of `user-agent` is a reasonable indicator of a first-party request? I was under the impression that programmatic consumers frequently set a user agent implicitly. Here's an example using `curl`:
```
~ curl localhost:9201/ -v
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 9201 (#0)
> GET / HTTP/1.1
> Host: localhost:9201
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< content-type: application/json; charset=UTF-8
< content-length: 538
<
```

Reply:
Well I'm not sure that all programmatic consumers behave like `curl`. I'm thinking along the likes of Node or Java, which (to my knowledge) don't set a user-agent header. That is an assumption on my part though, and I can't find any authoritative info in a brief search. I'll test this out tomorrow to verify.

That said, this header check is a best effort, I think false negatives are more acceptable than false positives. I'm all ears if you have a better idea for this though!

Reply:
> I'm thinking along the likes of Node or Java, which (to my knowledge) don't set a user-agent header

On their own, I don't _think_ the node's native `http`/`https` modules will send a user-agent, but it's also not very common for developers to build on these directly. Two of the most popular request libraries send a default `user-agent` header if you don't provide your own:

https://github.com/axios/axios/blob/e426910be7c417bdbcde9c18cb184ead826fc0e1/lib/adapters/http.js#L56-L61.

https://github.com/node-fetch/node-fetch/blob/a38b533ad68b333b46686034804c4b643de3b8d0/src/request.js#L182-L185

> That said, this header check is a best effort, I think false negatives are more acceptable than false positives.

I generally agree, but I worry that we'd be excluding so many requests that it would skew the usage stats.

> I'm all ears if you have a better idea for this though!

I wouldn't call them "better", but I have a couple of ideas:

1) We could introduce a new `kbn-fp` header (or similar) to indicate that a request is a "first party" request.
2) If we don't like the overhead an additional http header could introduce, then we could attach additional metadata to the existing `kbn-version` header. Instead of sending `7.11.0` as the value, we could send something like `7.11.0;fp`.

Of those two ideas, I like the first better. I don't think that an additional http header adds any significant overhead, and it should be relatively easy to add

Reply:
> I generally agree, but I worry that we'd be excluding so many requests that it would skew the usage stats.

Ah, thanks for pointing that out. Perhaps the best course of action is to just remove the `user-agent` check entirely, then, since we don't _need_ it. All of our first-party requests include `kbn-version` and `referer`, we could rely on that.

> Of those two ideas, I like the first better. I don't think that an additional http header adds any significant overhead, and it should be relatively easy to add

How do you feel about simply removing the check for `user-agent` now and revisiting this later?

In the vast majority of our API docs, we only instruct consumers to provide `xbn-xsrf`, not `kbn-version` -- the only exception that I can find is in [Reporting](https://www.elastic.co/guide/en/kibana/current/automating-report-generation.html#_use_a_script) (and tbh we should probably remove that in favor of `kbn-xsrf`!)

Reply:
> How do you feel about simply removing the check for user-agent now and revisiting this later?

I'm good with that approach too!

Reply:
> I'm good with that approach too!

Great, added an issue so we'll remember to revisit later: #85869